subprojects/core/src/test/groovy/org/gradle/internal/classpath/AccessTrackingPropertiesTest.groovy

```diff
@@ -203,6 +264,327 @@ class AccessTrackingPropertiesTest extends AbstractAccessTrackingMapTest {
         "load(InputStream)" | call(p -> p.load(new ByteArrayInputStream(new byte[0])))
     }
 
+    def "method put(#key, #newValue) reports argument as input and change"() {
+        when:
+        def result = getMapUnderTestToWrite().put(key, newValue)
+
+        then:
+        result == oldValue
+        1 * onAccess.accept(key, oldValue)
```

Comment:
❓ is this also testing the order in which the methods are called?

Reply:
It isn't, but the relative order of onAccess/onChange is important, so I've updated tests.

Reply:
Nice! I didn't know this pattern of splitting into multiple `then`s :+1: